lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -765,11 +817,46 @@ namespace Js
 
         SetLastIndexProperty(regEx, previousLastIndex, scriptContext);
 
-        return (JavascriptOperators::GetTypeId(result) == TypeIds_Null)
+        return JavascriptOperators::IsNull(result)
             ? TaggedInt::ToVarUnchecked(-1)
             : JavascriptOperators::GetProperty(RecyclableObject::FromVar(result), PropertyIds::index, scriptContext);
     }
 
+    Var JavascriptRegExp::EntrySymbolSplit(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(args, callInfo);
+        Assert(!(callInfo.Flags & CallFlags_New));
+
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        CHAKRATEL_LANGSTATS_INC_BUILTINCOUNT(RegexSymbolSplitCount);
+
+        RecyclableObject *thisObj = GetThisObject(args, L"RegExp.prototype[Symbol.match]", scriptContext);
+        JavascriptString* string = GetFirstStringArg(args, scriptContext);
+
+        // TODO: SPEC DEVIATION
+        //
+        // In RegexHelper::RegexSplit, we check if RegExp properties are overridden in order to determine
+        // if the algorithm is observable. If it is, we go through the new ES6 algorithm, but otherwise, we
+        // run the faster ES5 version.
+        //
+        // According to the spec, we're supposed to process "limit" after we use some of the RegExp properties.
+        // However, there doesn't seem to be any reason why "limit" processing can't be pulled above the rest
+        // in the spec. Therefore, we should see if such a spec update is OK. If not, this would have to be
+        // moved to its correct place in the code.
+        uint32 limit = (args.Info.Count < 3 || JavascriptOperators::IsUndefinedObject(args[2], scriptContext))
+            ? UINT_MAX
+            : JavascriptConversion::ToUInt32(args[2], scriptContext);
```

Comment:
@bterlson This is what I talked to you about regarding where the `limit` argument is processed.